vmsdk/python/tests/tdx/test_sdk.py

```diff
@@ -0,0 +1,107 @@
+"""TDX specific test."""
+
+import logging
+import re
+from hashlib import sha384
+from tests import test_sdk
+from cctrusted_base.tcg import TcgAlgorithmRegistry, TcgImrEvent
+from cctrusted_vm.sdk import CCTrustedVmSdk
+
+LOG = logging.getLogger(__name__)
+
+class TestCCTrustedVmSdkTdx(test_sdk.TestCCTrustedVmSdk):
+    """TDX specific test."""
+
+    MEASUREMENT_COUNT = 4
+    """Intel TDX TDREPORT provides the 4 measurement registers by default."""
+
+    PROC_CMDLINE= "/proc/cmdline"
+    """Linux kernel cmdline arguments."""
+
+    IMA_SUPPORTED_POLICIES = ["critical_data", "tcb", "fail_securely", ""]
+    """IMA supported policies.
+        When the policy is not specified or empty ("ima_policy="), the kernel
+    will measure boot_aggregate by default.
+    """
+
+    RTMR_LENGTH_BY_BYTES = 48
+    """RTMR length by bytes."""
+
+    def test_get_default_algorithms(self):
+        """Test default algorithm is supported."""
+        algo = CCTrustedVmSdk.inst().get_default_algorithms()
+        assert algo is not None
+        assert algo.alg_id == TcgAlgorithmRegistry.TPM_ALG_SHA384
+
+    def test_get_measurement_count(self):
+        """Test measurement count is 4 (RTMR count)."""
+        count = CCTrustedVmSdk.inst().get_measurement_count()
+        assert count == TestCCTrustedVmSdkTdx.MEASUREMENT_COUNT
+
+    def replay_eventlog(self):
+        """Get RTMRs from event log by replay."""
+        rtmr_len = TestCCTrustedVmSdkTdx.RTMR_LENGTH_BY_BYTES
+        rtmr_cnt = TestCCTrustedVmSdkTdx.MEASUREMENT_COUNT
+        rtmrs = [bytearray(rtmr_len)] * rtmr_cnt
+        event_logs = CCTrustedVmSdk.inst().get_eventlog().event_logs
+        assert event_logs is not None
+        for event in event_logs:
+            if isinstance(event, TcgImrEvent):
+                sha384_algo = sha384()
+                sha384_algo.update(rtmrs[event.imr_index] + event.digests[0].hash)
+                rtmrs[event.imr_index] = sha384_algo.digest()
+        return rtmrs
+
+    def check_imr(self, imr_index: int, alg_id: int, rtmr: bytes):
```

Comment:
If it is not a test case with the prefix of "test_", then please add "_" to indicate it as an internal function.